### ISSUE TITLE
[SUGGESTION] Fan control ?

### ISSUE THREAD

**Issue by mrwiwi** [Snikii/Lenovo-Thinkpad-X240#10]
x240 a plagued with a loud fan, on windows and linux you can use tools to control fan, and some thinkpad P users got control over with SSDT-CFAN table with clover.

Exemple Here : https://github.com/MirkoCovizzi/thinkpad-p51-hackintosh/blob/master/EFI/CLOVER/ACPI/patched/SSDT-CFAN.dsl

Another good example for clevo : https://github.com/FreeJHack/ClevoService

I wonder if it's possible to add such thing for us ?

I tried to add the patched SSDT file to OC (Can't see the table on maciasl, but i'm trying ^^), and the necessary ACPIPoller.kext, but i can't boot, will try later.

**Comment by ghost** [comment]
Yeah I'm in hope of someone adding this function too... This will save my ears. X240 fan is a disaster

**Comment by ghost** [comment]
[X240-SSDT-CFAN.zip](https://github.com/Sniki/Lenovo-Thinkpad-X240/files/4990714/X240-SSDT-CFAN.zip)

Please try this SSDT. I modded this SSDT from a X250 configuration and it works great under the OpenCore config with ACPIPoller.

**Comment by ghost** [comment]
> [X240-SSDT-CFAN.zip](https://github.com/Sniki/Lenovo-Thinkpad-X240/files/4990714/X240-SSDT-CFAN.zip)
> 
> Please try this SSDT. I modded this SSDT from a X250 configuration and it works great under the OpenCore config with ACPIPoller.

Using this SSDT might cause Battery status to be not working. Probably need a quick fix @Sniki would you mind helping us with that. Original SSDT link: https://github.com/qwerty12/Hackintosh/blob/c83d4c1237f7fcd96825974d5cf63e9e7ff53a84/Lenovo-ThinkPad-T450/SSDTs/SSDT-BATT.dsl

**Comment by ghost** [comment]
[FanControl_X240_BATC.zip](https://github.com/Sniki/Lenovo-Thinkpad-X240/files/4993063/FanControl_X240_BATC.zip)

Please replace the SSDT_BATC with the attached file, you then have a working fan control. Edit it with Rehabman MaciASL to change the temperature.
